batched/dense/unit_test/Test_Batched_SerialGemm.hpp

```diff
@@ -14,133 +14,133 @@
 
-  Kokkos::fill_random(a_expected, random, value_type(1.0));
-  Kokkos::fill_random(b_expected, random, value_type(1.0));
-  Kokkos::fill_random(c_expected, random, value_type(1.0));
+  ScalarType randStart, randEnd;
+  KokkosKernels::Impl::getRandomBounds(1.0, randStart, randEnd);
+  Kokkos::fill_random(A, rand_pool, randStart, randEnd);
+  Kokkos::fill_random(B, rand_pool, randStart, randEnd);
+  Kokkos::fill_random(C, rand_pool, randStart, randEnd);
 
-  Kokkos::fence();
-
-  Kokkos::deep_copy(a_actual, a_expected);
-  Kokkos::deep_copy(b_actual, b_expected);
-  Kokkos::deep_copy(c_actual, c_expected);
+  Kokkos::deep_copy(C_ref, C);
 
   Functor_BatchedVanillaGEMM<ViewType, ViewType, ViewType, execution_space> vgemm;
-  vgemm.A_t = std::is_same<transA, Trans::Transpose>::value;
-  vgemm.B_t = std::is_same<transB, Trans::Transpose>::value;
-  vgemm.A_c = vgemm.B_c = false;
-  vgemm.A               = a_expected;
-  vgemm.B               = b_expected;
-  vgemm.C               = c_expected;
-  vgemm.alpha           = alpha;
-  vgemm.beta            = beta;
-  vgemm.run();  // Compute c_expected
-  Functor_TestBatchedSerialGemm<DeviceType, ViewType, ScalarType, ParamTagType, AlgoTagType>(alpha, a_actual, b_actual,
-                                                                                             beta, c_actual)
-      .run();
-
-  typename ViewType::HostMirror c_expected_host = Kokkos::create_mirror_view(c_expected);
-  typename ViewType::HostMirror c_actual_host   = Kokkos::create_mirror_view(c_actual);
-
-  // Copy to host for comparison
-  Kokkos::deep_copy(c_expected_host, c_expected);
-  Kokkos::deep_copy(c_actual_host, c_actual);
-
-  Kokkos::fence();
-
-  // check c_expected = c_actual
-  // std::conditional<, float,
+  vgemm.A_t   = !std::is_same_v<transA, KokkosBatched::Trans::NoTranspose>;
+  vgemm.B_t   = !std::is_same_v<transB, KokkosBatched::Trans::NoTranspose>;
+  vgemm.A_c   = std::is_same_v<transA, KokkosBatched::Trans::ConjTranspose>;
+  vgemm.B_c   = std::is_same_v<transB, KokkosBatched::Trans::ConjTranspose>;
+  vgemm.A     = A;
+  vgemm.B     = B;
+  vgemm.C     = C_ref;
+  vgemm.alpha = alpha;
+  vgemm.beta  = beta;
+  vgemm.run();  // Compute C_ref
+
+  // Compute using gemm API
+  auto info =
+      Functor_TestBatchedSerialGemm<DeviceType, ViewType, ScalarType, ParamTagType, AlgoTagType>(alpha, A, B, beta, C)
+          .run();
+  EXPECT_EQ(info, 0);
+
+  auto h_C     = Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), C);
+  auto h_C_ref = Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), C_ref);
+
+  // check C = C_ref
   using mag_type = typename ats::mag_type;
   mag_type sum(1), diff(0);
 
   mag_type eps = ats::epsilon();
-
-  eps *= std::is_same<value_type, Kokkos::Experimental::half_t>::value ||
-                 std::is_same<value_type, Kokkos::Experimental::bhalf_t>::value
+  eps *= std::is_same_v<ValueType, Kokkos::Experimental::half_t> ||
```

Comment:
This definition of epsilon worries me a bit, we should revisit to actually compute the expected accuracy based on the compute parameters chosen for the test...